alerts/cluster-monitoring-operator/PrometheusKubernetesListWatchFailures.md

```diff
@@ -0,0 +1,58 @@
+# PrometheusKubernetesListWatchFailures
+
+## Meaning
+
+The `PrometheusKubernetesListWatchFailures` alert is triggered when there is a constant
+increase in failures with `LIST/WATCH` requests to the Kubernetes API during Prometheus target
+discovery.
+
+## Impact
+
+This may prevent Prometheus from adding, updating, or deleting targets effectively.
+
+## Diagnosis
+
+Determine whether the alert has triggered for the instance of Prometheus used
+for default cluster monitoring or for the instance that monitors user-defined
+projects by viewing the alert message's `namespace` label: the namespace for
+default cluster monitoring is `openshift-monitoring` and the namespace for
+user workload monitoring is `openshift-user-workload-monitoring`.
+
+## Mitigation
+
+To gain further insight, review the logs of the affected Prometheus instance:
+
+   ```shell
+   $ NAMESPACE='<value of namespace label from alert>'
+
+   $ oc -n $NAMESPACE logs -c prometheus -l 'app.kubernetes.io/name=prometheus'
```

Comment:
FYI, this command only output 20-line logs, maybe the error is not in the output, maybe
`$ oc -n $NAMESPACE logs -c prometheus ${prometheus_pod}`
is better?

Reply:
I meant the command only output 20-line logs, the error maybe not in the 20-line logs
`oc -n $NAMESPACE logs -c prometheus -l 'app.kubernetes.io/name=prometheus' --tail=-1` will show all logs